src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -52,6 +52,23 @@ public FileTextLoaderOptionsProvider()
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
     public class FileTextLoader : TextLoader
     {
+        /// <summary>
+        /// Guards access to <see cref="_workspace"/> and <see cref="_lazyTextAndVersion"/>.
+        /// </summary>
+        private readonly object _gate = new object();
+
+        /// <summary>
+        /// This is the workspace used to initialize <see cref="_lazyTextAndVersion"/>.
```

Comment:
@tmat Is this workspace parameter something that is now deprecated more or less for us, since this was being used to report errors but you have since cleaned that up?

Reply:
The workspace gets passed as an argument to a public API.

Reply:
It is deprecated, but unfortunately also gets passed to public API :(.